metricbeat/module/vsphere/virtualmachine/virtualmachine.go

```diff
@@ -104,110 +123,174 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
+						"total_size": virtualMachine.Summary.Config.MemorySizeMB,
+						"reserved":   virtualMachine.Summary.Config.MemoryReservation,
+					},
+					"power_state":                   virtualMachine.Summary.Runtime.PowerState,
+					"snapshot_consolidation_needed": virtualMachine.Summary.Runtime.ConsolidationNeeded,
+					"vmx_path":                      virtualMachine.Summary.Config.VmPathName,
 				}
+
+				// add mapped values
+				resourcePool := rpMap[virtualMachine.ResourcePool.Value]
+				event.Put("resource_pool.name", resourcePool.Name)
+				event.Put("resource_pool.path", resourcePool.InventoryPath)
+
+				for _, value := range metric.Value {
+					switch series := value.(type) {
+					case *types.PerfMetricIntSeries:
+						counter := counterMap[series.Id.CounterId]
+						event.Put(counter, series.Value)
+					case *types.PerfMetricSeriesCSV:
+						counter := counterMap[series.Id.CounterId]
+						event.Put(counter, series.Value)
+					default:
+						m.Logger().Debug("Metric is of an unknown type, skipping")
+					}
+				}
+
+				// Get host information for VM
+				if host := virtualMachine.Summary.Runtime.Host; host != nil {
+					event["host"] = mapstr.M{
+						"id": host.Value,
+					}
+					hostSystem, err := getHostSystem(ctx, c, host.Reference())
+					if err == nil {
+						event.Put("host.hostname", hostSystem.Summary.Config.Name)
+					} else {
+						m.Logger().Debug(err.Error())
+					}
+				} else {
+					m.Logger().Debug("'Host', 'Runtime' or 'Summary' data not found. This is either a parsing error " +
+						"from vsphere library, an error trying to reach host/guest or incomplete information returned " +
+						"from host/guest")
+				}
+
+				// Get custom fields (attributes) values if get_custom_fields is true.
+				if m.GetCustomFields && virtualMachine.Summary.CustomValue != nil {
+					customFields := getCustomFields(virtualMachine.Summary.CustomValue, customFieldsMap)
+
+					if len(customFields) > 0 {
+						event["custom_fields"] = customFields
+					}
+				} else {
+					m.Logger().Debug("custom fields not activated or custom values not found/parse in Summary data. This " +
+						"is either a parsing error from vsphere library, an error trying to reach host/guest or incomplete " +
+						"information returned from host/guest")
+				}
+
+				if virtualMachine.Summary.Vm != nil {
+					networkNames, err := getNetworkNames(ctx, c, virtualMachine.Summary.Vm.Reference())
+					if err != nil {
+						m.Logger().Debug(err.Error())
+					} else {
+						if len(networkNames) > 0 {
+							event["network_names"] = networkNames
+						}
+					}
+				}
+
+				if virtualMachine.Datastore != nil {
+					var datastoreIds []string
+					for _, datastore := range virtualMachine.Datastore {
+						datastoreIds = append(datastoreIds, datastore.Value)
+					}
+					event["datastore.id"] = datastoreIds
+				}
+
+				reporter.Event(mb.Event{
+					MetricSetFields: event,
+				})
```

Comment:
Looks cleaner.

```suggestion
				reporter.Event(mb.Event{MetricSetFields: event})
```

Reply:
thanks, I'm used to python/black and formatting options like that not really being a thing I can adjust, I'll make sure to pay more attention to that kind of thing moving forward when writing go.

Reply:
you can use gofumpt. stricter gofmt (and gofmt compliant). it'd help (better than gofmt).